src/ir_passes/await.ml

```diff
@@ -68,12 +81,13 @@ let rec t_async context exp =
    (* add the implicit return label *)
    let k_ret = fresh_cont (typ exp1) T.unit in
    let k_fail = fresh_err_cont T.unit in
+   let k_clean = fresh_cont T.unit T.unit in
    let context' =
```

Comment:
Should context' perhaps contain a Cleanup label? Maybe that's why your other test isn't chaining as expected across await* calls... Not sure.

Reply:
Yes, ae686d6d5. I also just noticed the asymmetry.